pentasharp/Controllers/AddTaxiDriverController.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Models.Enums;
+using WebApplication1.Filters;
+using pentasharp.Data;
+using pentasharp.Services;
+using pentasharp.Models.TaxiRequest;
+
+namespace WebApplication1.Controllers
+{
+    [Route("Business/TaxiDriver")]
+    public class AddTaxiDriverController : Controller
+    {
+        private readonly ITaxiService _taxiService;
+        private readonly IDriverService _driverService;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly AppDbContext _context;
+
+        public AddTaxiDriverController(
+            ITaxiService taxiService,
+            IDriverService driverService,
+            IHttpContextAccessor httpContextAccessor,
+            AppDbContext context)
+        {
+            _taxiService = taxiService;
+            _driverService = driverService;
+            _httpContextAccessor = httpContextAccessor;
+            _context = context;
+        }
+
+        [HttpPost("AddDriver")]
+        public async Task<IActionResult> AddDriver([FromBody] RegisterDriverRequest model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(new { success = false, message = "Invalid data provided." });
+            }
+
+            var userId = GetCompanyId();
```

Comment:
services should have all the logic and be only called in controllers, so move the logic from this endpoint and all others added, to the designated service AddDriverAsync